pypeit/scripts/show_1dspec.py

```diff
@@ -35,34 +35,47 @@ def main(args, unit_test=False):
     from pypeit import specobjs
     from pypeit import msgs
 
-    from IPython import embed
-
-    sobjs = specobjs.SpecObjs.from_fitsfile(args.file)
-
-    # List only?
-    if args.list:
-        print("Showing object names for input file...")
-        for ii in range(len(sobjs)):
-            name = sobjs[ii].name
-            print("EXT{:07d} = {}".format(ii+1, name))
-        return
-
-    # Load spectrum
-    if args.obj is not None:
-        exten = sobjs.name.index(args.obj)
-        if exten < 0:
-            msgs.error("Bad input object name: {:s}".format(args.obj))
-    else:
-        exten = args.exten-1 # 1-index in FITS file
-
-    # Check Extraction
-    if args.extract == 'OPT':
-        if 'OPT_WAVE' not in sobjs[exten]._data.keys():
-            msgs.error("Spectrum not extracted with OPT.  Try --extract=BOX")
+    try:
```

Comment:
I think the try except is not unreasonable here. The issue is that we need to be able to display specobjs files as well as coadded spectra which have a different data model (and will always have to). A cleaner solution would be to have the header indicate what kind of file this is, i.e. specobjs vs coadd. I do not like @kbwestfall  suggestion of  having a command line argument, because why make the user type extra stuff when  it is trivial to figure out what kind of file we are dealing with (from i.e. a header card or whatever).

Reply:
Fine, but the possible input for `file` needs to be clear in the help string of the argument.  Leave it for another PR.